core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -642,74 +621,34 @@ public long getTableLastUpdatedTime() {
   }
 
   /**
-   * to get the number of dimension present in the table
-   *
-   * @param tableName
-   * @return number of dimension present the table
-   */
-  public int getNumberOfDimensions(String tableName) {
-    return tableDimensionsMap.get(tableName).size();
-  }
-
-  /**
-   * to get the number of allMeasures present in the table
-   *
-   * @param tableName
-   * @return number of allMeasures present the table
-   */
-  public int getNumberOfMeasures(String tableName) {
-    return tableMeasuresMap.get(tableName).size();
-  }
-
-  /**
-   * to get the all dimension of a table
-   *
-   * @param tableName
-   * @return all dimension of a table
-   */
-  public List<CarbonDimension> getDimensionByTableName(String tableName) {
-    return tableDimensionsMap.get(tableName);
-  }
-
-  /**
-   * to get the all measure of a table
-   *
-   * @param tableName
-   * @return all measure of a table
-   */
-  public List<CarbonMeasure> getMeasureByTableName(String tableName) {
-    return tableMeasuresMap.get(tableName);
-  }
-
-  /**
-   * Return all dimensions of the table
+   * Return all visible dimensions of the table
    */
   public List<CarbonDimension> getDimensions() {
```

Comment:
Why didn't use  getVisibleDimensions instead of getDimensions?  There is getAllDimensions, it's confused